src/main/java/bio/terra/service/snapshot/flight/create/CreateSnapshotFirecloudGroupStep.java

```diff
@@ -0,0 +1,60 @@
+package bio.terra.service.snapshot.flight.create;
+
+import bio.terra.service.auth.iam.IamService;
+import bio.terra.service.auth.iam.exception.IamConflictException;
+import bio.terra.service.auth.iam.exception.IamNotFoundException;
+import bio.terra.service.snapshot.flight.SnapshotWorkingMapKeys;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import bio.terra.stairway.exception.RetryException;
+
+public class CreateSnapshotFirecloudGroupStep implements Step {
+  private final IamService iamService;
+
+  public CreateSnapshotFirecloudGroupStep(IamService iamService) {
+    this.iamService = iamService;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    String groupName =
+        context
+            .getWorkingMap()
```

Comment:
This method retrieves `context.getWorkingMap()` twice, I would do it once at the start
```
    var workingMap = context.getWorkingMap();
```